### PR TITLE
Feature/Task-13 CREATED DIRECTORIES AND  DTO FOR ENTITIES 

### PR DESCRIPTION
**Pull request by 2musime**
## CREATING IMPORTANT DIRECTORIES AND  DTO FOR ENTITIES
This PR introduces DTOs for the Certificate, Supplier, User, and Comment entities. 

### Description
This PR introduces the Data Transfer Objects (DTOs) for the following entities in the project:
- Certificate
- Supplier
- User
- Comment
-  AssignedUser

### Changes: Files created    

- [x]     `CertificateDto`: Contains certificate details.
- [x]     `SupplierDto`: Contains supplier details.
- [x]     `UserDto`: Contains user details. 
- [x]     `CommentDto`: Contains comment details.
- [x]    `CertificateCreateDto`
- [x]    `CertificateUpdateDto`
- [x] `AssignedUserDto`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -253,6 +253,7 @@ src/.vs/config/applicationhost.config
 .idea/
 
 # AppSetting.json
-appsettings.Development.json
 
+appsettings.Development.json
+appsettings.json
```
Comment (elhasic) [inline .gitignore:258]: appsettings.json should not be ignored

File: backend/CertificateManagerAPIs/DTO/CertificateCreateDto.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        public SupplierDto Supplier { get; set; } = null!;
+        public string Type { get; set; } = null!;
+        public DateTime ValidFrom { get; set; }
+        public DateTime ValidTo { get; set; }
+        public string PdfFile { get; set; } = null!;
+        public int? UserAssigned { get; set; }
+        public int SupplierId { get; set; }
+    }
+
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/DTO/CertificateCreateDto.cs]: unnecessary new line

File: backend/CertificateManagerAPIs/DTO/CertificateDto.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿namespace CertificateManagerAPIs.DTO
+{
+    public class CertificateDto
+    {
+
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/DTO/CertificateDto.cs]: unnecessary new line

File: backend/CertificateManagerAPIs/DTO/CertificateDto.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        public SupplierDto Supplier { get; set; } = null!;
+        public string Type { get; set; } = null!;
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/DTO/CertificateDto.cs:6]: you have to decide will you have new line between properties or not, you can not mix

File: backend/CertificateManagerAPIs/DTO/CertificateDto.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+        public DateTime ValidFrom { get; set; }
+
+        public DateTime ValidTo { get; set; }
+        public int Id { get; set; }
+    }
+
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/DTO/CertificateDto.cs]: unnecessary new line

File: backend/CertificateManagerAPIs/DTO/CertificateUpdateDto.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    public class CertificateUpdateDto
+    {
+        public string Type { get; set; } = null!;
+        public DateTime ValidFrom { get; set; }
+        public DateTime ValidTo { get; set; }
+        public string PdfFile { get; set; } = null!;
+        public int? UserAssigned { get; set; }
+        public int SupplierId { get; set; }
+    }
+
```
Comment (elhasic) [inline backend/CertificateManagerAPIs/DTO/CertificateUpdateDto.cs]: unnecessary new line

File: backend/CertificateManagerAPIs/Scripts/202409251028_certificatedb_create_tables.sql
Comment (elhasic) [inline backend/CertificateManagerAPIs/Scripts/202409251028_certificatedb_create_tables.sql:1]: you cannot repeat scripts in each of this files, each file have to have only scripts that is not executed before in order how should be executed

**Review by elhasic — CHANGES_REQUESTED** [review]

File: backend/CertificateManagerAPIs/Scripts/202409251028_certificatedb_create_tables.sql
Comment (2musime) [reply backend/CertificateManagerAPIs/Scripts/202409251028_certificatedb_create_tables.sql:1]: I did get this well, You mean that I `Scripts` are to be run only once, or is it to be run on each branch?
Please help me to know that.
Thank you.

**Review by elhasic — APPROVED** [review]

**Pull request merged**